lib/statefileio.c

```diff
@@ -110,48 +111,72 @@ sf_FIST_delay_on_write();
 //
 
 
-extern int
-is_drbd_device(
+enum {
+    DEVICE_TYPE_NORMAL = 0,
+    DEVICE_TYPE_DRBD = 1,
+    DEVICE_TYPE_NBD = 2
+};
+
+static int
+get_device_type(
       const char *path)
 {
     static const int drbd_major = 147;
+    static const int nbd_major = 43;
 
     struct stat stats;
     if (stat(path, &stats) == -1)
     {
         log_internal(MTC_LOG_ERR, "SF: failed to execute stat (sys %d).\n", errno);
-        return 0;
+        return DEVICE_TYPE_NORMAL;
```

Comment:
shouldn't you just return an error here ? Either a DEVICE_TYPE_ERROR or "-errno"  for coherency?